products/identityplatform/api.yaml

```diff
@@ -0,0 +1,408 @@
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'clientSecret'
+        description: |
+          The client secret of the OAuth client, to enable OIDC code flow.
+  - !ruby/object:Api::Resource
+    name: 'TenantOauthIdpConfig'
+    base_url: 'projects/{{project}}/tenants/{{tenant}}/oauthIdpConfigs'
+    self_link: 'projects/{{project}}/tenants/{{tenant}}/oauthIdpConfigs/{{name}}'
+    create_url: 'projects/{{project}}/tenants/{{tenant}}/oauthIdpConfigs?oauthIdpConfigId={{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      OIDC IdP configuration for a Identity Toolkit project within a tenant.
+
+      You must enable the
+      ((Google Identity Platform)[https://console.cloud.google.com/marketplace/details/google-cloud-platform/customer-identity])
+      in the marketplace prior to using this resource.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The name of the OauthIdpConfig. Must start with `oidc.`.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'tenant'
+        required: true
+        url_param_only: true
+        description: |
+          The name of the tenant where this OIDC IDP configuration resource exists
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        description: |
+          Human friendly display name.
+      - !ruby/object:Api::Type::Boolean
+        name: 'enabled'
+        description: |
+          If allows users to sign in with the provider.
+      - !ruby/object:Api::Type::String
+        name: 'issuer'
+        description: |
+          For OIDC Idps, the issuer identifier.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'clientId'
+        description: |
+          The client id of an OAuth client.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'clientSecret'
+        description: |
+          The client secret of the OAuth client, to enable OIDC code flow.
+  - !ruby/object:Api::Resource
+    name: 'Tenant'
+    base_url: 'projects/{{project}}/tenants'
+    self_link: 'projects/{{project}}/tenants/{{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      Tenant configuration in a multi-tenant project.
+
+      You must enable the
+      ((Google Identity Platform)[https://console.cloud.google.com/marketplace/details/google-cloud-platform/customer-identity])
+      in the marketplace prior to using this resource.
+
+      You must ((enable multi-tenancy)[https://cloud.google.com/identity-platform/docs/multi-tenancy-quickstart])
+      via the Cloud Console prior to creating tenants.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The name of the tenant that is generated by the server
+        output: true
+      - !ruby/object:Api::Type::String
```

Comment:
Are really none of the fields in this resource required?

Reply:
Ah, `displayName` is required, I'll mark it as such.

This resource is a bit strange and uses the `displayName` as a base for the generated `name` field